common/packet/src/packet.cpp

```diff
@@ -0,0 +1,225 @@
+#include "packet.hpp"
+
+/**
+ * Packet class
+ * Write and Read - byte, int, float, double, glm::vec3
+ * Read returns -1 on error, size of data read (in bytes) otherwise
+ * 
+ * Note: currently no support for checking if read in the wrong order
+ *      or if read type mismatches from original written value
+*/
+
+union Union32 {
+    float f;
+    int i;
+    u_int32_t l;
+} union32;
+
+union Union64 {
+    double f;
+    u_int64_t l;
+} union64;
+
+void Packet::write_byte(char data){
+    u_int8_t toAdd = u_int8_t(data);
+    buffer.push_back(toAdd);
```

Comment:
```suggestion
    buffer.push_back((uint8_t) data);
```
The function already makes a copy of the parameter and puts it on the stack, no need to make another copy.